### PR TITLE
build: upgrade to Python 3.9

### LINKED ISSUE #590: Upgrade to at least Python version 3.8
Apart from the Walrus operator, there are some nice new features. We could also ditch `attrs` and use `dataclasses` instead!

### PR DESCRIPTION
**Pull request by eddieantonio**
Resolves #590

### CODE REVIEW

**Comment by eddieantonio** [comment]
Still draft because I need to setup Python 3.9 on Sapir to make this work!

File: docs/production-on-sapir.md [hunk 1]
```diff
@@ -1,13 +1,16 @@
 Production on Sapir
 -------------------
 
-As of <time datetime="2019-11-18">November 18, 2019</time>, the Cree
-Dictionary is deployed on a server called **Sapir**. You can visit the
+As of <time datetime="2021-01-05">January 5, 2021</time>, the itwêwina
```
Comment (andrewdotn) [inline docs/production-on-sapir.md:4]: Is `the itwêwina` correct? I’d been thinking of the project name as a proper noun.

File: docs/production-on-sapir.md [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
+is deployed on a server called **Sapir**. You can visit the
 production website here:
 
 <https://sapir.artsrn.ualberta.ca/cree-dictionary>
 
 
+It was supposed to be deprecated in 2020 and decommissions in 2021, but
```
Comment (andrewdotn) [inline docs/production-on-sapir.md:11]: s/(decommission)s/\1ed/

File: CreeDictionary/CreeDictionary/settings.py [hunk 3]
```diff
@@ -271,52 +267,21 @@
         "django.contrib.staticfiles.storage.ManifestStaticFilesStorage"
     )
 
-log_dir = Path(BASE_DIR) / "django_logs"
-log_dir.mkdir(exist_ok=True)
-
```
Comment (andrewdotn) [inline CreeDictionary/CreeDictionary/settings.py:276]: Normally I really like having persistent logs on disk, but wow is

sapir:/data/texts/opt/cree-intelligent-dictionary/CreeDictionary/django_logs/django.log

ever noisy with all sorts of ValueError / AttributeError exceptions, and no
timestamps :s When running under systemd, console logs will at least be timestamped.

*note that systemd logs are by default cleared on reboot*

File: docs/production-on-sapir.md (hunk 1 above)
Comment (eddieantonio) [reply docs/production-on-sapir.md:4]: Good catch!

File: CreeDictionary/CreeDictionary/settings.py (hunk 3 above)
Comment (eddieantonio) [reply CreeDictionary/CreeDictionary/settings.py:276]: Thanks! I redirected to console and made a new issue: #596

**Pull request merged**